term/fhir-term/src/main/java/com/ibm/fhir/term/service/FHIRTermService.java

```diff
@@ -1167,29 +1166,77 @@ private List<FHIRTermServiceProvider> loadProviders() {
+     *     the outcome of validation; true if either the lookupOutcome or coding parameters are null or have a null display
+     * @implNote this method looks up the implied CodeSystem from the FHIR registry to determine case sensitivity,
+     *     then delegates to {@link #validateDisplay(Coding, LookupOutcome, boolean)}
+     */
+    private ValidationOutcome validateDisplay(Coding coding, LookupOutcome lookupOutcome) {
+        if (coding == null) {
+            return buildValidationOutcome(true, null, lookupOutcome);
+        }
+
+        java.lang.String system = (coding.getSystem() != null) ? coding.getSystem().getValue() : null;
+        java.lang.String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : null;
+
+        boolean caseSensitive = false;
+        if (system != null) {
+            java.lang.String url = (version != null) ? system + "|" + version : system;
+            caseSensitive = CodeSystemSupport.isCaseSensitive(url);
+        }
+
+        return validateDisplay(coding, lookupOutcome, caseSensitive);
+    }
+
+    /**
+     * Validate a coding display against the passed lookupOutcome
+     *
+     * @param coding
+     *     the coding
+     * @param lookupOutcome
+     *     the outcome of a CodeSystem lookup
+     * @param isCaseSensitive
+     *     whether the underlying CodeSystem is case-sensitive or not
+     * @return
+     *     the outcome of validation; true if either the lookupOutcome or coding parameters are null or have a null display
+     */
+    private ValidationOutcome validateDisplay(Coding coding, LookupOutcome lookupOutcome, boolean isCaseSensitive) {
         if (lookupOutcome == null || coding == null ||
                 lookupOutcome.getDisplay() == null || coding.getDisplay() == null ||
                 lookupOutcome.getDisplay().getValue() == null && coding.getDisplay().getValue() == null) {
             return buildValidationOutcome(true, null, lookupOutcome);
         }
 
-        java.lang.String system = null;
-        if (coding.getSystem() != null) {
-            system = coding.getSystem().getValue();
-        } else if (codeSystem != null && codeSystem.getUrl() != null) {
-            system = codeSystem.getUrl().getValue();
+        java.lang.String displayToValidate = isCaseSensitive ? coding.getDisplay().getValue() : normalize(coding.getDisplay().getValue());
+
+        boolean result = false;
+        if (isCaseSensitive) {
+            result = lookupOutcome.getDisplay().getValue().equals(displayToValidate);
+            if (result == false) {
+                result = lookupOutcome.getDesignation().stream()
+                        .anyMatch(d -> d.getValue() != null && displayToValidate.equals(d.getValue().getValue()));
+            }
+        } else {
+            result = normalize(lookupOutcome.getDisplay().getValue()).equals(displayToValidate);
+            if (result == false) {
+                result = lookupOutcome.getDesignation().stream()
```

Comment:
comment as to what this is doing would be useful

Reply:
```suggestion
                // call it valid if any of the normalized designations matches the normalized display value
                result = lookupOutcome.getDesignation().stream()
```